lib/Conversion/VerifToSMT/VerifToSMT.cpp

```diff
@@ -386,8 +388,10 @@ struct VerifBoundedModelCheckingOpConversion
               // Create an ITE to calculate the next reg state
               // TODO: we create a lot of ITEs here that will slow things down -
               // these could be avoided by making init/loop regions concrete
-              nextRegStates.push_back(builder.create<smt::IteOp>(
-                  loc, isPosedge, regInput, regState));
+              nextRegStates.push_back(
```

Comment:
We could also skip all the operation building above on rising clock mode here, e.g., by factoring it out into a lambda and returning the value that should be added to 'nextRegStates' and returning early if risingClockOnly.

Reply:
Realised that in the rising clocks case we actually only need to do an append with the reg inputs, so I've wrapped it in an if statement that I think is a bit cleaner than a lambda.